### PR TITLE
Remove illegal nested properties

### PR DESCRIPTION
**Pull request by ACMoretxj**
* Removed illegal repeat nested properties in `Resource/MessageHeader`.
* Removed illegal repeat nested properties in `Resource/Organization`.
* Updated test resources.

P.S.

Some of `Resource/MessageHeader` and `Resource/Provenance` will have its ID changed. Because we have escapped the MSH segments in the subsequent commits.

### CODE REVIEW

File: data/Templates/Hl7v2/ADT_A01.liquid [hunk 1]
```diff
@@ -5,9 +5,6 @@
         {% assign firstSegments = hl7v2Data | get_first_segments: 'PID|PD1|PV1|PV2|AVR|MSH' -%}
 
         {% evaluate messageHeaderId using 'ID/MessageHeader' MSH: firstSegments.MSH -%}
-        {% evaluate organizationId4 using 'ID/Organization' HD: firstSegments.MSH.4 -%}
-        {% evaluate organizationId6 using 'ID/Organization' HD: firstSegments.MSH.6 -%}
-        {% evaluate organizationId22 using 'ID/Organization' XON: firstSegments.MSH.22 -%}
```
Comment (BoyaWu10) [inline data/Templates/Hl7v2/ADT_A01.liquid:10]: Are we removing organizations from MSH segment?

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/TestData/Expected/Hl7v2/ADT_A01/ADT01-28-expected.json [hunk 2]
```diff
@@ -1,66 +1,27 @@
...
-                  }
-                ]
-              }
+              "code": "P",
+              "system": "http://terminology.hl7.org/CodeSystem/v2-0103"
             }
           ]
-        },
-        "sender": {
-          "reference": "Organization/2d7d77cd-0ca7-13a5-c9ed-6e70b739adb4"
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/TestData/Expected/Hl7v2/ADT_A01/ADT01-28-expected.json:63]: Why the change influences the generation of `sender`?

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/TestData/Expected/Hl7v2/ADT_A01/ADT04-251-expected.json [hunk 3]
```diff
@@ -1,30 +1,24 @@
-﻿{
+{
   "resourceType": "Bundle",
   "type": "batch",
   "entry": [
     {
-      "fullUrl": "urn:uuid:7393bcca-6d2f-543d-7787-0753133a8496",
+      "fullUrl": "urn:uuid:55e2d2b6-8c08-6f34-9f2c-ccf2ba785dba",
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/TestData/Expected/Hl7v2/ADT_A01/ADT04-251-expected.json:6]: Is this change caused by previous changes? It seems this change won't change the ID.

Comment (ACMoretxj) [reply src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/TestData/Expected/Hl7v2/ADT_A01/ADT04-251-expected.json:6]: Because the first generation of the ID is very long ago, we have escapped the `MSH` value in subsequent commits.

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/TestData/Expected/Hl7v2/ADT_A01/ADT01-28-expected.json (hunk 2 above)
Comment (ACMoretxj) [reply src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/TestData/Expected/Hl7v2/ADT_A01/ADT01-28-expected.json:63]: The `sender` is actually an `Organization`, which we have deleted.

File: data/Templates/Hl7v2/ADT_A01.liquid (hunk 1 above)
Comment (ACMoretxj) [reply data/Templates/Hl7v2/ADT_A01.liquid:10]: Yes, the `MSH` and its related resources(especially Organization) seems contain some issues in mapping tables. So we should remove them currently and fix them in `v3.4` or later releases.

**Review by qiwjin — APPROVED** [review]

**Review by BoyaWu10 — APPROVED** [review]

**Review by moria97 — APPROVED** [review]
